lib/pure/os.nim

```diff
@@ -23,6 +23,19 @@ when defined(windows):
   import winlean
 elif defined(posix):
   import posix
+
+  proc toTime(ts: Timespec): times.Time {.inline.} =
+    result = initTime(ts.tv_sec.int64, ts.tv_nsec.int)
+
+  # Some systems have Stat.st_*time (second resolution) instead of
+  # Stat.st_*tim (nanosecond resolution).
+  const StatHasNano = compiles(Stat.st_atim)
```

Comment:
hrm, I think you can just use ``declared`` here. If you can't, you should use the relevant ``define`` instead of ``compiles``.